### DISCUSSION TITLE
Can I use existing PVC as a disk source?

### DISCUSSION THREAD

**Discussion by stevefan1999-personal** [longhorn/longhorn#5927] [Q&A]
I want to use AKS with Longhorn but the disk volume is created and managed by the CSI plugin, which is another PVC from another storage class. I wonder if I can just get it provisioned reliably, so that when I have node maintenance events the pods that relies on Longhorn are correctly migrated. 

I happen to have tried a hail-marry approach where I modified the storage class and make it mounted preemptively rather than on-demand, hence Longhorn indirectly migrated the disks by observing new disk event in all the nodes. While it kinda works, it is not reliable.

**Comment by AdamJSoftware** [comment]
I am also wondering about this

**Comment by DamiaSan** [comment]
Maybe you can have a look at https://github.com/longhorn/longhorn/discussions/2190
